tests/client/test_redirects.py

```diff
@@ -243,12 +249,29 @@ class MockDispatch(AsyncDispatcher):
 
 
 async def test_body_redirect(backend):
+    """
+    A 308 redirect should preserve the request body.
+    """
     client = AsyncClient(dispatch=MockDispatch(), backend=backend)
     url = "https://example.org/redirect_body"
     data = b"Example request body"
     response = await client.post(url, data=data)
     assert response.url == URL("https://example.org/redirect_body_target")
-    assert response.json() == {"body": "Example request body"}
+    assert response.json()["body"] == "Example request body"
+    assert "content-length" in response.json()["headers"]
+
+
+async def test_no_body_redirect(backend):
+    """
+    A 303 redirect should remove the request body.
```

Comment:
Should we test we have the same behavior for 302 as well?

Reply:
No need to over-egg it. 😀
Test Cases have a maintenance overhead too, and it’s not functionally different right?

Reply:
I mean, it kind of is another functional path from my point of view, but I agree it might not be worth it considering the things we’d have to setup (another extremely similar handler on the test app, and a copy of this test with 302 instead). Let’s leave it. :)

Reply:
Is it not possible to parametrize the test case?